### ISSUE TITLE
Dashboard not loading variables values properly.

### ISSUE THREAD

**Issue by heliapb** [grafana/grafana#46995]
**What happened**:
Our team has a couple of dashboards that are loaded to grafana, to multiple environments And for those different environments the  value for the variables is going to be different. And since we are creating the dashboards and the same dashboards are used on all of those environments, so far we have been able to have the dashboard working properly, by removing any value from a variable and changing the select option to false, as shown bellow:
```
(...)
  "templating": {
    "list": [
      {
        "allFormat": "pipe",
        "allValue": ".+",
        "current": {
          "selected": false,
          "text": "",
          "value":  ""
        },
        "datasource": "prometheus",
        "definition": "label_values(up{job=\"cassandra\"}, availability_set)",
```
We also have select the query on dasboard load:
![Screenshot from 2022-03-28 14-21-40](https://user-images.githubusercontent.com/66432275/160412541-48fccbef-e27a-4533-9646-895194d987c9.png)

And like this each time we open the same dashboard on all the environments it was loading the correct variables for that environments with no issue. But since we upgraded grafana to the version from v7.5.6 to v8.4.3 we stated getting this weird error:

 ![grafana](https://user-images.githubusercontent.com/66432275/160413274-07f961b1-6ab5-4132-b984-a46bbb3323bc.gif)
Were it stopped to load a value to the variable. And since for this case we cannot use the all option for this variable we are faced with this issue.
**What you expected to happen**:
To have the same behavior as before, as by letting the value for the variable empty, for the dashboard to load one and not to break

**How to reproduce it (as minimally and precisely as possible)**:
By having a dashboard created on the v7.5.6, by making a similar  change to the variable values, and then updating the grafana version.
**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.4.3
- Data source type & version: prometheus v2.31.0
- OS Grafana is installed on: kubernetes v1.22.0
- User OS & Browser: n/a
- Grafana plugins: n/a
- Others: n/a

**Comment by torkelo** [comment]
You will have to provide more detail. 

> Were it stopped to load a value to the variable. And since for this case we cannot use the all option for this variable we are faced with this issue.
What you expected to happen:
To have the same behavior as before, as by letting the value for the variable empty, for the dashboard to load one and not to break

I fail to understand this sentance. Can you elaborate in more detail?

**Comment by heliapb** [comment]
Of course. The strategy of removing the values previous worked because when we added the dashboard to the various environments, it would load a value to the variable in the dashboard load. And when you when back to the dashboard folder and go back again to the dashboard, it would always load a value. But since we updated the grafana it basically stated to not to do that. Please let me know if you need more information. Thanks

**Comment by heliapb** [comment]
@torkelo Don't know if you had the chance to review the previous information. But if you please have the chance it would be much appreciated. Thanks

**Comment by heliapb** [comment]
Hi there, we just upgraded to grafana 9.0.4 and the issue still happens. If somebody could look into this, thanks.

**Comment by pauljohnston2009** [comment]
I just updated to 9.2.0-77716pre (3864de9425) to fix another issue, and am also experiencing something similar to this on some of my dashboards.

I have one dashboard where variables are dependant on eachother, loaded from prometheus datasource, which seems to work fine.
Another dashboard has 2 mysql datasource variables, not dependant on eachother. When I change the order of the variables the last one loads correct each time, the first does not load.

**Comment by pauljohnston2009** [comment]
can also reproduce on new dashboard, with new variables
both variables configured as follows
![image](https://user-images.githubusercontent.com/21235196/188828567-09c03a02-c98c-47e3-873d-031bcda26ba3.png)

when one is first
![image](https://user-images.githubusercontent.com/21235196/188828186-bdaff361-d2f4-413b-9665-a0b4db3519c9.png)
![image](https://user-images.githubusercontent.com/21235196/188828358-f066ca80-f892-4bd5-9fa8-a64baf41f273.png)
![image](https://user-images.githubusercontent.com/21235196/188828425-3c73b19c-e7ad-4a11-b438-21b9486a9207.png)


when one is second
![image](https://user-images.githubusercontent.com/21235196/188828882-1b8fe476-49b8-4198-accc-d0b67a694b9b.png)
![image](https://user-images.githubusercontent.com/21235196/188828735-320ad4fe-bf87-4cc2-9ece-8bb438749f80.png)
![image](https://user-images.githubusercontent.com/21235196/188828825-caa6b48a-e7cf-47f2-822e-6f052b34da6f.png)

**Comment by pauljohnston2009** [comment]
It looks like the outstanding request gets cancelled when the next variables query starts. Perhaps because the refId is the same?
looking at an older version of grafana the refId use to be the name of the variable, it seems to be hard coded now.

![image](https://user-images.githubusercontent.com/21235196/189006664-8cedc475-7eee-4c43-9768-58a73dfa7728.png)
![image](https://user-images.githubusercontent.com/21235196/189006720-a658c4b7-7aa5-40b2-b012-0034d450e659.png)
![image](https://user-images.githubusercontent.com/21235196/189006754-b301e890-5146-4c94-b8c4-dad436dc27aa.png)

**Comment by Beshelmek** [comment]
> It looks like the outstanding request gets cancelled when the next variables query starts. Perhaps because the refId is the same? looking at an older version of grafana the refId use to be the name of the variable, it seems to be hard coded now.
> 
> ![image](https://user-images.githubusercontent.com/21235196/189006664-8cedc475-7eee-4c43-9768-58a73dfa7728.png) ![image](https://user-images.githubusercontent.com/21235196/189006720-a658c4b7-7aa5-40b2-b012-0034d450e659.png) ![image](https://user-images.githubusercontent.com/21235196/189006754-b301e890-5146-4c94-b8c4-dad436dc27aa.png)

Still literally same problem on 9.2.0 release

@torkelo look at this

**Comment by jeremy-smith-maco** [comment]
> > It looks like the outstanding request gets cancelled when the next variables query starts. Perhaps because the refId is the same? looking at an older version of grafana the refId use to be the name of the variable, it seems to be hard coded now.
> > ![image](https://user-images.githubusercontent.com/21235196/189006664-8cedc475-7eee-4c43-9768-58a73dfa7728.png) ![image](https://user-images.githubusercontent.com/21235196/189006720-a658c4b7-7aa5-40b2-b012-0034d450e659.png) ![image](https://user-images.githubusercontent.com/21235196/189006754-b301e890-5146-4c94-b8c4-dad436dc27aa.png)
> 
> Still literally same problem on 9.2.0 release
> 
> @torkelo look at this

Appears to be fixed in 9.2.2

**Comment by zuchka** [comment]
> Appears to be fixed in 9.2.2

Can confirm that this appears fixed on our public sandbox, which is currently on 9.2.5. Check out the dashboard here:

https://play.grafana.org/d/9BWltAd4k/46995?orgId=1

Closing this for now as it appears fixed. Happy to reopen if it is not. If so, please provide clear steps for reproduction 👍
